R/create_demographic_lookup.R

```diff
@@ -0,0 +1,420 @@
+create_demographic_lookup <- function(data, year, write_to_disk = TRUE) {
+  check_variables_exist(
+    data,
+    c(
+      "chi", "cij_marker", "recid", "diag1", "diag2", "diag3", "diag4", "diag5", "diag6", "age", "sigfac", "spec",
+      "dementia", "hefailure", "refailure", "liver", "cancer", "cvd", "copd", "chd", "parkinsons", "ms",
+      "epilepsy", "asthma", "arth", "diabetes", "atrialfib", "cost_total_net",
+      "deathdiag1", "deathdiag2", "deathdiag3", "deathdiag4", "deathdiag5",
+      "deathdiag6", "deathdiag7", "deathdiag8", "deathdiag9", "deathdiag10",
+      "deathdiag11"
+    )
+  )
+
+  demo_lookup <- data %>%
+    # Remove missing chi
+    dplyr::filter(!is_missing(.data$chi)) %>%
+    # Add the various cohorts
+    dplyr::mutate(
+      mh = assign_mh_cohort(recid, diag1, diag2, diag3, diag4, diag5, diag6),
+      frail = assign_frailty_cohort(recid, diag1, diag2, diag3, diag4, diag5, diag6, spec, sigfac),
+      maternity = assign_maternity_cohort(recid),
+      high_cc = assign_high_cc_cohort(dementia, hefailure, refailure, liver, cancer, spec),
+      medium_cc = assign_medium_cc_cohort(cvd, copd, chd, parkinsons, ms),
+      low_cc = assign_low_cc_cohort(epilepsy, asthma, arth, diabetes, atrialfib),
+      comm_living = assign_comm_living_cohort(),
+      adult_major = assign_adult_major_condition_cohort(recid, age, cost_total_net),
+      child_major = assign_child_major_condition_cohort(recid, age, cost_total_net),
+      end_of_life = assign_eol_cohort(
+        recid, deathdiag1, deathdiag2, deathdiag3, deathdiag4, deathdiag5,
+        deathdiag6, deathdiag7, deathdiag8, deathdiag9, deathdiag10,
+        deathdiag11
+      )
+    ) %>%
+    assign_substance_cohort() %>%
+    # Aggregate to CHI level
+    dplyr::group_by(.data$chi) %>%
+    dplyr::summarise(dplyr::across(c(
+      "mh",
+      "frail",
+      "maternity",
+      "high_cc",
+      "medium_cc",
+      "low_cc",
+      "comm_living",
+      "adult_major",
+      "child_major",
+      "end_of_life",
+      "substance"
+    ), any)) %>%
+    dplyr::ungroup() %>%
+    # Assign demographic_cohort based on hierarchy of each cohort
+    dplyr::mutate(demographic_cohort = dplyr::case_when(
+      end_of_life ~ "End of Life",
+      frail ~ "Frailty",
+      high_cc ~ "High Complex Conditions",
+      maternity ~ "Maternity and Healthy Newborns",
+      mh ~ "Mental Health",
+      substance ~ "Substance Misuse",
+      medium_cc ~ "Medium Complex Conditions",
+      low_cc ~ "Low Complex Conditions",
+      child_major ~ "Child Major Conditions",
+      adult_major ~ "Adult Major Conditions",
+      comm_living ~ "Assisted Living in the Community",
+      TRUE ~ "Healthy and Low User"
+    )) %>%
+    # Reorder variables
+    dplyr::relocate(.data$demographic_cohort, .after = chi)
+
+  # Write to disk
+  if (write_to_disk == TRUE) {
+    write_rds(demo_lookup,
+      path = glue::glue("{get_slf_dir()}/Cohorts/Demographic_Cohorts_{year}.rds")
```

Comment:
Create a function for this (unless it already exists - possibly make existing rds)

Reply:
function already called `get_demog_cohorts_path` in rds format